src/lib/ucan.ts

```diff
@@ -11,70 +9,63 @@ export type Validation = {
   validProofs: boolean
 }
 
-export const decode = (token: string): Ucan | null => {
+export const decode = async (token: string): Promise<Ucan | null> => {
   try {
-    return ucan.decode(token)
+    return await ucan.validate(token, { checkIsExpired: false, checkIsTooEarly: false, checkSignature: false })
```

Comment:
Will this only fail when a UCAN can't be decoded? 

If not, maybe we want to implement a decode function in this project. Otherwise we might show the wrong error to the user.

Reply:
Yes this will only fail if the UCAN can't be decoded :+1:

Reply:
I wouldn't recommend repeating the decode functionality. The decode logic is getting quite a lot more complicated with upcoming compatibility changes (supporting old UCAN versions).

Reply:
Maybe there was a misunderstanding in a discussion last year. I can see myself having said something like "we'll do the decoding ourselves".
What I meant was that we wouldn't use the `Chained` type from ts-ucan, but instead would do the recursive decoding of a ucan and then its proofs ourself (which we're doing).

Reply:
Yep, that's what I understood. 👍 

And that sounds good, let's go with this. Just wanted to make sure decoding was the only failure case here. 👌